### PR TITLE
Use generics in RunCursorQueryForSchema

### PR DESCRIPTION
**Pull request by janisz**
## Description

Use generics. 

Look at 
- [ ] `pkg/search/postgres/common.go`
- [ ] `tools/generate-helpers/pg-table-bindings/store.go.tpl`

## Testing Performed

CI

### CODE REVIEW

**Comment by janisz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #3446** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/3446" target="_blank">Graphite</a> 
    * **PR #3448** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/3448" target="_blank">Graphite</a> 
      * **PR #3449** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/3449" target="_blank">Graphite</a> 
        * **PR #3450** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/3450" target="_blank">Graphite</a> 
          * **PR #3451** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/3451" target="_blank">Graphite</a> 
            * **PR #3452** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/3452" target="_blank">Graphite</a> 
              * **PR #3453** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/3453" target="_blank">Graphite</a>  👈
                * **PR #3454** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/3454" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/3453?utm_source=stack-comment).

**Comment by janisz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #3446** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/3446" target="_blank">Graphite</a> 
    * **PR #3448** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/3448" target="_blank">Graphite</a> 
      * **PR #3449** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/3449" target="_blank">Graphite</a> 
        * **PR #3450** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/3450" target="_blank">Graphite</a> 
          * **PR #3451** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/3451" target="_blank">Graphite</a> 
            * **PR #3452** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/3452" target="_blank">Graphite</a> 
              * **PR #3453** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/3453" target="_blank">Graphite</a>  👈
                * **PR #3454** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/3454" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/3453?utm_source=stack-comment).

File: pkg/search/postgres/common.go [hunk 1]
```diff
@@ -897,6 +897,66 @@ func RunCursorQueryForSchema(ctx context.Context, schema *walker.Schema, q *v1.Q
 	}, closer, nil
 }
 
+// RunCursorQueryForSchemaType creates a cursor against the database
+func RunCursorQueryForSchemaType[T any, PT interface {
+	Unmarshal([]byte) error
+	*T
+}](ctx context.Context, schema *walker.Schema, q *v1.Query, db *pgxpool.Pool) (fetcher func(n int) ([]*T, error), closer func(), err error) {
+	query, err := standardizeQueryAndPopulatePath(q, schema, GET)
```
Comment (misberner) [inline pkg/search/postgres/common.go:905]: There's a lot of code duplication here. Can you at least factor out the cursor creation to a common function?

**Review by misberner — CHANGES_REQUESTED** [review]

File: pkg/search/postgres/common.go (hunk 1 above)
Comment (janisz) [reply pkg/search/postgres/common.go:768]: Extracted common code

**Review by misberner — APPROVED** [review]

**Pull request merged**